### PR TITLE
Redesign

### PR DESCRIPTION
**Pull request by JeTSkY1h**
title Says it all

### CODE REVIEW

**Review by Droggelbecher92 — DISMISSED** [review]
lgtm 👌

File: backend/src/main/java/com/github/JeTSkY1h/book/Book.java [hunk 1]
```diff
@@ -21,6 +21,11 @@ public class Book {
     List<String> genre;
     String coverPath;
 
+    Integer rating;
+
+    Integer rated;
+
+
```
Comment (gossie) [inline backend/src/main/java/com/github/JeTSkY1h/book/Book.java]: Fields should be final.

File: backend/src/main/java/com/github/JeTSkY1h/book/BookController.java [hunk 2]
```diff
@@ -21,6 +22,15 @@
 public class BookController {
     private final BookService bookService;
 
+    @PutMapping("/api/book/rate/{rating}")
```
Comment (gossie) [inline backend/src/main/java/com/github/JeTSkY1h/book/BookController.java]: URLs should identify resources. The rating is not the resource, the book is. I would use `/api/books/{id}` as the URL and would use it for all changes that happen to books.

File: backend/src/main/java/com/github/JeTSkY1h/book/BookController.java [hunk 3]
```diff
@@ -21,6 +22,15 @@
...
+    ResponseEntity<Book> rateBook(@RequestBody Book ratedbook, @PathVariable Integer rating) {
+        return ResponseEntity.of(Optional.of(bookService.rateBook(ratedbook, rating)));
+    }
+
+    @GetMapping(("/{id}/book"))
```
Comment (gossie) [inline backend/src/main/java/com/github/JeTSkY1h/book/BookController.java]: I would use `/api/books/{id}` as URL.

File: backend/src/main/java/com/github/JeTSkY1h/book/Book.java (hunk 1 above)
Comment (JeTSkY1h) [reply backend/src/main/java/com/github/JeTSkY1h/book/Book.java]: could you please explain further why?

Comment (Droggelbecher92) [reply backend/src/main/java/com/github/JeTSkY1h/book/Book.java]: He wanted to write private. package-private ist still accessible from the same package, which conflicts with encapsulation.

**Review by Droggelbecher92 — APPROVED** [review]
Lgtm 👍  Could implement Andrés suggestions (Id-Pathvariables at the end, visibility modifiers in your Models)

**Pull request merged**